### PR TITLE
Sync ref docs

### PR DESCRIPTION
**Pull request by mdlinville**
@dnephin @thaJeztah PTAL. This is all the files we accidentally edited in docker.github.io repository since the merge.

### CODE REVIEW

File: docs/reference/api/docker_remote_api_v1.18.md
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.18.md]: These two lines should be reverted

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: see https://github.com/docker/docker/pull/27434

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: see https://github.com/docker/docker/pull/26516 for here and other places

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: see https://github.com/docker/docker/commit/690882c2e79c3f3742c709cf158584e61594ba00

**Comment by mstanleyjones** [comment]
Hold fire, this should have been based on 1.12x, not master! Need to rebase and it's a little complicated. I'll ping when it's ready.

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: Same

Comment (thaJeztah) [inline docs/reference/builder.md]: Also looks like most other changes in this document should be reverted

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: this is a fix; can stay in

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: This can stay in as well

File: docs/reference/commandline/cli.md
Comment (thaJeztah) [inline docs/reference/commandline/cli.md]: This should be reverted

File: docs/reference/commandline/cli.md
Comment (thaJeztah) [inline docs/reference/commandline/cli.md]: same here

File: docs/reference/commandline/cli.md
Comment (thaJeztah) [inline docs/reference/commandline/cli.md]: needs to be reverted as well

File: docs/reference/commandline/dockerd.md
Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md]: This as well

File: docs/reference/commandline/dockerd.md
Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md]: needs to be reverted https://github.com/docker/docker/pull/26212

File: docs/reference/commandline/dockerd.md
Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md]: Looks like a fix, this can stay

File: docs/reference/commandline/dockerd.md
Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md]: This can be removed https://github.com/docker/docker/pull/27432

File: docs/reference/commandline/dockerd.md
Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md]: This, and other changes below need to be reverted

File: docs/reference/commandline/inspect.md
Comment (thaJeztah) [inline docs/reference/commandline/inspect.md]: Document needs to be reverted as well (other than the metadata, and `{% raw %}`)

File: docs/reference/commandline/node_inspect.md
Comment (thaJeztah) [inline docs/reference/commandline/node_inspect.md]: Bit confused by this message; what's the backward-incompatibility (as this command didn't exist before); maybe this originally referred to it being an `RC` / preview, so changes could still occur?

File: docs/reference/commandline/node_inspect.md
Comment (thaJeztah) [inline docs/reference/commandline/node_inspect.md]: Should be reverted

File: docs/reference/commandline/node_ps.md
Comment (thaJeztah) [inline docs/reference/commandline/node_ps.md]: Changes in this document should be reverted as well

File: docs/reference/commandline/ps.md
Comment (thaJeztah) [inline docs/reference/commandline/ps.md]: Same for this document (other than meta/raw)

File: docs/reference/commandline/service_inspect.md
Comment (thaJeztah) [inline docs/reference/commandline/service_inspect.md]: Changes in this document should be reverted (other than meta/raw)

File: docs/reference/commandline/service_ps.md
Comment (thaJeztah) [inline docs/reference/commandline/service_ps.md]: Changes in this document should be reverted (other than meta/raw/links)

File: docs/reference/commandline/service_scale.md
Comment (thaJeztah) [inline docs/reference/commandline/service_scale.md]: typo; `immediatly`

File: docs/reference/commandline/stack_tasks.md
Comment (thaJeztah) [inline docs/reference/commandline/stack_tasks.md]: This command no longer exists (is now `stack ps`)

File: docs/reference/commandline/version.md
Comment (thaJeztah) [inline docs/reference/commandline/version.md]: Should be reverted

File: docs/reference/commandline/volume_inspect.md
Comment (thaJeztah) [inline docs/reference/commandline/volume_inspect.md]: Changes in this document should be reverted (other than meta/raw/links)

File: docs/reference/commandline/volume_ls.md
Comment (thaJeztah) [inline docs/reference/commandline/volume_ls.md]: Should be reverted https://github.com/docker/docker/pull/25720

Comment (thaJeztah) [inline docs/reference/commandline/volume_ls.md]: Same for other changes in this document

File: docs/reference/run.md
Comment (thaJeztah) [inline docs/reference/run.md]: Should be reverted

File: docs/reference/run.md
Comment (thaJeztah) [inline docs/reference/run.md]: Should be reverted

File: docs/reference/run.md
Comment (thaJeztah) [inline docs/reference/run.md]: Looks like this was a fix, so can be kept https://github.com/docker/docker.github.io/commit/e41a76bf8199558cc4c4c8bf856a65c0dea8c079

**Review by thaJeztah — CHANGES_REQUESTED** [review]
hm, looks like this is going to be tough to sync :cry:

**Comment by thaJeztah** [comment]
oh, LOL, long-live GitHub cache

**Comment by mstanleyjones** [comment]
OK, this should be rebased now. Let's see how it looks.

File: docs/reference/api/docker_remote_api_v1.18.md
Comment (mstanleyjones) [inline docs/reference/api/docker_remote_api_v1.18.md]: Yeah it should work without the .md file...

File: docs/reference/api/docker_remote_api_v1.24.md
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.24.md]: @mstanleyjones ^^

File: docs/reference/api/docker_remote_api_v1.25.md
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.25.md]: Same here :smile:

File: docs/reference/commandline/service_scale.md
Comment (thaJeztah) [inline docs/reference/commandline/service_scale.md]: ^^ :smile:

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Only a few left, but much better than the previous changeset 😄

File: docs/reference/builder.md
Comment (mstanleyjones) [inline docs/reference/builder.md]: What are you referring to? The typo fix or the line wrapping change?

File: docs/reference/commandline/dockerd.md
Comment (mstanleyjones) [inline docs/reference/commandline/dockerd.md]: This was just reformatting the JSON to be less indented, I think.

File: docs/reference/api/docker_remote_api_v1.24.md
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.24.md]: Looks like it was cherry-picked; https://github.com/docker/docker/pull/26960

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: Oh, yes, this is ok, no need to change anything

File: docs/reference/builder.md
Comment (thaJeztah) [inline docs/reference/builder.md]: Oh, yes, this is ok as well, no need to change anything

**Review by dnephin — APPROVED** [review]
LGTM

**Review by thaJeztah — COMMENTED** [review]
changes LGTM, but thinking again if we should update the 1.12.x branch with these changes as the 1.12.x branch has the "old style" docs in it (so full documentation). Changing this is would no longer allow building those docs from the 1.12.x branch.

Sorry for rambling, trying to get my head around it :D

**Comment by thaJeztah** [comment]
Basically, question is, should we change to "new style" docs in the current release branch (as it's a "breaking" change in what could be considered a "patch" release)

**Comment by mstanleyjones** [comment]
Yuck, you are right. I don't know the answer. The old docs in v1.12 branch are stranded anyway -- we can't build them to put on docs.docker.com at all anymore. If we leave them alone that means that we can't really touch them at all anymore and therefore can't improve them on docs.docker.com. It also means that the changes @dnephin wants to do will only be available in master.

Whether we take this PR or not, I should try to apply this same changeset to master and resolve any problems, so that we can move forward.

**Comment by dnephin** [comment]
Oops, you're right. I didn't notice this was a PR against 1.12.x. I think we need the PR against master.

**Comment by mstanleyjones** [comment]
@dnephin just to remind you, master on docker.github.io is equivalent to 1.12x on docker/docker. For master on docker/docker you need vnext-engine on docker.github.io. So I don't think we have changed much with the docs on the vnext-engine branch at all. The changes that were upsetting you were in master. Anyway I am working on a PR to get these changes into docker/docker master so we can benefit from typo correction etc. Let's see how hard it is to resolve the conflicts.

**Comment by dnephin** [comment]
I'm not really concerned with what is about to be published, I'm concerned with the "single source of truth" for the reference docs. That is the master branch of docker/docker.  Anything in the docker.github.io repo under the reference section should be considered disposable, regardless of the branch. The problem is we have changes there that are not up-streamed to docker/docker master.

> The changes that were upsetting you were in master.

I don't see any evidence of that. For examples the meta-data on this page: https://raw.githubusercontent.com/docker/docker/master/docs/reference/api/docker_remote_api.md is not the same as the metadata being added in this PR. It looks like someone made a related, but not identical change to this repo.

My concern is that I need to make significant structural changes to the reference doc files. If there are any oustanding changes to these files they will be near impossible to apply after my changes are merged. Or my PR will heavily conflict with any large changes (like the one to the metadata).

I'm trying to avoid that pain by making sure that all of the change that were applied directly in the docs repo are applied here first, so we don't lose them.

**Comment by thaJeztah** [comment]
So yes, I think we should apply the TOML -> YAML changes and the fixes for the new Markdown generator to master. 

Possibly that would be something like; on docker.github.io rebase master onto vnext-engine, and apply that to docker/master?

**Pull request closed without merging**
